xtask/src/runchecks.rs

```diff
@@ -313,18 +327,12 @@ fn main() {
     // are run.
     //
     // If no environment has been passed, run all checks.
-    match env::args()
-        .nth(
-            1, /* Index of the first argument, because 0 is the binary name */
-        )
-        .as_deref()
-    {
-        Some("std") => std_checks(),
-        Some("no_std") => no_std_checks(),
-        Some("typos") => check_typos(),
-        Some("examples") => check_examples(),
-        Some(other) => panic!("Unexpected test type: {}", other),
-        None => {
+    match env {
+        CheckType::Std => std_checks(),
+        CheckType::NoStd => no_std_checks(),
+        CheckType::Typos => check_typos(),
+        CheckType::Examples => check_examples(),
+        CheckType::All => {
```

Comment:
I fear this has reintroduced the problem I tried to solve with https://github.com/burn-rs/burn/pull/762/commits/824cd6ae384adbc29e587b559ab37b6334ebfcd1, where any invalid input is treated as "all"

Reply:
I think we can return an error if another kind of command has been inserted, what do you think @DrChat and @dae?

Reply:
Already cover, since it uses clap, it's validated when parsing the input:

```bash
$ cargo xtask run-checks tt
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/xtask run-checks tt`
error: invalid value 'tt' for '<ENV>'
  [possible values: all, std, no-std, typos, examples]

For more information, try '--help'.
```